falcon/app.py

```diff
@@ -841,6 +920,7 @@ def handler(req, resp, ex, params):
                 )
 
         # TODO(vytas): Remove this shimming in a future Falcon version.
+        assert handler is not None
```

Comment:
Is this validation needed?
IIRC the best practice is not to rely on `assert` in production code (since it is, e.g., possible to silence `assert`s). If we want to validate `handler` here, we could explicitly write, `if handler is None: raise ValueError(...)`. We could also check if it is callable, not just `not None`.

Also, I think it is better to perform this validation just before my `TODO`, because that shimming is starts with `get_argnames`.

Reply:
iirc it was to avoid typing errors. I can remove it and see if I have another way of handling the typing issues

Reply:
I mean, that assert doesn't hurt, and we can even leave it inplace if needed, but it would be better to start off with raising a `ValueError`. In fact, I think it would hurt to validate `handler` this way.